bundles/framework/statehandler/session-methods.js

```diff
@@ -47,15 +50,37 @@ Oskari.clazz.category(
                 me.setSessionExpiring(minutes);
                 popup.close(true);
             });
+            logoutButton.setTitle(logoutButtonMessage);
+            logoutButton.setHandler(function () {
+                jQuery('#loginbar').find("form[action='/logout']").submit();
+            });
 
             return {
                 show: function () {
-                    popup.show(notifyTitle, notifyMessage, [extendButton]);
-
-                    expireTimeout = setTimeout(function () {
-                        popup.show(expiredTitle, expiredMessage);
-                        popup.makeModal();
-                    }, 60 * 1000);
+                    const expireIn = 60; // Expire time in seconds
+                    popup.show(notifyTitle, notifyMessage + '<br />' + expiresInMessage.replace('{expires}', (expireIn)), [logoutButton, extendButton]);
```

Comment:
`Oskari.getMsg()` implements string interpolation. No need to use `.replace(..)`. Same goes for `notifyMessage` on line 40.